outlook/src/taskpane/components/Login/Login.tsx

```diff
@@ -23,98 +35,189 @@ class Login extends React.Component<{}, LoginState> {
         this.state = {
             isLoading: false,
             baseURL: localStorage.getItem("baseURL"),
-            urlError: null
+            authenticationRequestError: AuthenticationRequestError.None,
+            isCheckingUrl: false,
         };
     }
 
   onServerChange = (_, newValue?: string): void => {
         this.setState({
             baseURL: newValue,
-            urlError: null
+            authenticationRequestError: AuthenticationRequestError.None,
+            isCheckingUrl: false,
         });
     }
 
     sanitizeDbUrl = (urlStr: string): string => {
-        let tmpStr = urlStr;
-        // Without the protocol URL won't parse it.
-        if (!tmpStr.startsWith('http://') && !tmpStr.startsWith('https://')) {
-            tmpStr = 'https://' + tmpStr;
+        try {
+            let tmpStr = urlStr;
+            // Without the protocol URL won't parse it.
+            if (!tmpStr.startsWith('http://') && !tmpStr.startsWith('https://')) {
+                tmpStr = 'https://' + tmpStr;
+            }
+            const url = new URL(tmpStr);
+            url.protocol = 'https:';
+            return url.origin;
+        } catch {
+            return null;
         }
-        const url = new URL(tmpStr);
-        url.protocol = 'https:';
-        return url.origin;
     }
 
-    login = () => {
-        let sanitizedURL = '';
-        try {
-            sanitizedURL = this.sanitizeDbUrl(this.state.baseURL);
-        } catch (err) {
-            this.setState({urlError: 'The database URL is invalid. It should follow the scheme \'https://mycompany.odoo.com\''})
+    login = async () => {
+        const sanitizedURL = this.sanitizeDbUrl(this.state.baseURL);
+        if (!sanitizedURL) {
+            this.setState({isCheckingUrl: false, authenticationRequestError: AuthenticationRequestError.InvalidScheme});
             return;
         }
 
-        //const context = React.useContext(AppContext);
+        this.setState({
+            baseURL: sanitizedURL,
+            isCheckingUrl: true,
+            authenticationRequestError: AuthenticationRequestError.None,
+        });
+
         api.baseURL = sanitizedURL;
-        localStorage.setItem('baseURL', sanitizedURL)
+        localStorage.setItem('baseURL', sanitizedURL);
+
+        if (!await this._isOdooDatabaseReachable()) {
```

Comment:
Don't think you need to await here since the method returns a boolean?

Reply:
The method is asynchronous so we need to await the result

Reply:
(if a method call "await" it would return automatically a promise)

Reply:
Yes I had a feeling that was the reason when writing my comment.
It's weird to me but I guess it's JS...

Reply:
It's the same in python :) But yes in JS it's even more strange (you can mix async and non-async function)
```python
import asyncio

async def a():
    return 3

async def main():
    res = await a()
    print(res)

asyncio.run(main())
```